### PR TITLE
Add detailed network info to the device information endpoint

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
Addresses https://github.com/appium/appium/issues/12502

### CODE REVIEW

File: app/src/main/java/io/appium/uiautomator2/handler/GetDeviceInfo.java [hunk 1]
```diff
@@ -38,6 +45,51 @@ public GetDeviceInfo(String mappedUri) {
         super(mappedUri);
     }
 
+    private static JSONArray networksInfoToJSON(DeviceInfoHelper deviceInfoHelper,
```
[the remark this answers is not in the document]
Comment (mykola-mokhnach) [reply app/src/main/java/io/appium/uiautomator2/handler/GetDeviceInfo.java]: changed

File: app/src/main/java/io/appium/uiautomator2/handler/GetDeviceInfo.java [hunk 2]
```diff
@@ -38,6 +43,50 @@ public GetDeviceInfo(String mappedUri) {
...
+                        networkCaps.getLinkDownstreamBandwidthKbps());
+                caps.put("signalStrength",
+                        getField("mSignalStrength", networkCaps));
+                caps.put("networkSpecifier",
+                        formatNull(getField("mNetworkSpecifier", networkCaps)));
+                caps.put("SSID", formatNull(getField("mSSID", networkCaps)));
+            }
+            resultItem.put("capabilities", formatNull(networkCaps == null ? null : caps));
+
+            if (networkCaps != null || networkInfo != null) {
```
Comment (dpgraham) [inline app/src/main/java/io/appium/uiautomator2/handler/GetDeviceInfo.java]: Could we check that `resultItem` is not empty?

`if (resultItem.length() > 0) { ... }`

**Comment by dpgraham** [comment]
I'd like to add a test for this in `appium-uiautomator2-driver` when this is published.

File: app/src/main/java/io/appium/uiautomator2/handler/GetDeviceInfo.java (hunk 2 above)
Comment (mykola-mokhnach) [reply app/src/main/java/io/appium/uiautomator2/handler/GetDeviceInfo.java]: changed

**Comment by mykola-mokhnach** [comment]
I've added an integration test for getInfo endpoint for this module

**Pull request merged**